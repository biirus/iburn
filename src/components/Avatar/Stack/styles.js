export default ({ spacing, colors }) => {
  return {
    /* Styles applied to root element */
    root: {
      display: "flex",
      lineHeight: 1,
    },

    /* Styles applied to the wrapper of `Avatar` element */
    avatarWrapper: {
      position: "relative",
    },

    /* Styles applied to each `Avatar` element */
    avatar: {
      border: `2px solid ${colors.n0}`,
    },
  };
};
