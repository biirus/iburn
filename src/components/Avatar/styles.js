export default ({ colors, typography, spacing, }) => {
  const size = spacing.unit * 6;

  return {
    root: {
      ...typography.h600,
      color: colors.w700a,

      display: "inline-flex",
      alignItems: "center",
      justifyContent: "center",
      alignSelf: "baseline",
      overflow: "hidden",
      width: size,
      height: size,

      backgroundColor: colors.n100,
      borderRadius: "50%",
    },

    img: {
      width: "100%",
      height: "100%",
      objectFit: "cover",
    },
  };
};
