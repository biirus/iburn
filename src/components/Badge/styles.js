export default ({ spacing, palette, colors, borderRadius }) => {
  return {
    /* Styles applied to the root element */
    root: {
      display: "inline-flex",
      padding: [[spacing.small / 2, spacing.small]],
      lineHeight: 1,
      borderRadius,
    },

    /* Styles applied to the root element if `color="default"` */
    default: {
      backgroundColor: colors.n20a,
      color: colors.n400,
    },

    /* Styles applied to the root element if `color="primary"` */
    primary: {
      backgroundColor: palette.primary,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="error"` */
    error: {
      backgroundColor: palette.error,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="success"` */
    success: {
      backgroundColor: palette.success,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="warning"` */
    warning: {
      backgroundColor: palette.warning,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="info"` */
    info: {
      backgroundColor: palette.info,
      color: palette.text.invert,
    },
  };
};
