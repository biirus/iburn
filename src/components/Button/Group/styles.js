export default ({ borderRadius, }) => ({
  root: {
    display: "inline-flex",
  },

  justified: {
    width: "100%",

    "& $item": {
      flex: "1 0 auto",
      width: 0,

      "& $inner": {
        width: "100%",
        textAlign: "center",
      },
    },
  },

  item: {
    flex: "1 0 auto",

    "& + $item": {
      marginLeft: 1,
    },

    "&:not(:first-child):not(:last-child) $inner": {
      borderRadius: 0,
    },

    "&:first-child $inner": {
      borderRadius: [borderRadius, 0, 0, borderRadius,],
    },

    "&:last-child $inner": {
      borderRadius: [0, borderRadius, borderRadius, 0,],
    },
  },

  inner: {},
});
