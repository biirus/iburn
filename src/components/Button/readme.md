### Views

```js
const Wrapper = props => (
  <div
    style={{
      display: "grid",
      gridTemplateColumns: "repeat(3, max-content)",
      gridGap: "16px"
    }}
  >
    {props.children}
  </div>
);

<Wrapper>
  <Button>Default Button</Button>
  <Button view="thin">Thin Button</Button>
  <Button view="link">Link Button</Button>
</Wrapper>;
```

### Sizes

#### Inline

`Inline` size helps you to align a text arounded buttons. Typically it has a `view="thin"` or `view="link"` prop.

```js
<p>
  The button inside paragraph{" "}
  <Button size="inline" color="primary">
    Inline Button
  </Button>. This is example of button along with text (button is colored blue just
  for better visibility).
</p>
```

#### Block

```js
<Button size="block" color="warning">
  Block button occupies full width of it's parrent
</Button>
```

#### Compact

It is a regular button but it's paddings are reduced a bit.

```js
<Button size="compact" color="primary">
  Compact Button
</Button>
```

### Colors

If `view` prop equals `thin` or `link` then `color` prop will be ignored

```js
const Wrapper = props => (
  <div
    style={{
      display: "grid",
      gridTemplateColumns: "repeat(5, max-content)",
      gridGap: "16px"
    }}
  >
    {props.children}
  </div>
);

<Wrapper>
  <Button>Default Button</Button>
  <Button color="primary">Primary Button</Button>
  <Button color="warning">Warning Button</Button>
  <Button color="success">Success Button</Button>
  <Button color="error">Error Button</Button>
  <Button color="info">Info Button</Button>
</Wrapper>;
```

### Icons

You can use any icon before or after button label. It can be set only icon w/o label.

```js
const Wrapper = props => (
  <div
    style={{
      display: "grid",
      gridTemplateColumns: "repeat(3, max-content)",
      gridGap: "16px"
    }}
  >
    {props.children}
  </div>
);

<Wrapper>
  <Button icon={<Icon label="calendar" />}>Icon before</Button>
  <Button iconAfter={<Icon label="chevron-down" />}>Icon after</Button>
  <Button iconAfter={<Icon label="search" />} />
</Wrapper>;
```

### Component

Button has a special `component` prop. By using it you can specify any component to render button. Eg. react-router's `Link` or `NavLink`.
If will pass a `href` prop to button, then `<a>` tag will be rendered.

```js
const Wrapper = props => (
  <div
    style={{
      display: "grid",
      gridTemplateColumns: "repeat(3, max-content)",
      gridGap: "16px"
    }}
  >
    {props.children}
  </div>
);

<Wrapper>
  <Button component="div" color="primary">
    This is a div
  </Button>
  <Button href="https://google.com" target="_blank">
    Google.com
  </Button>
</Wrapper>;
```
