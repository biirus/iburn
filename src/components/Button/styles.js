import createDefaultStyle from "./lib/colors/default";
import createPrimaryStyle from "./lib/colors/primary";
import createWarningStyle from "./lib/colors/warning";
import createSuccessStyle from "./lib/colors/success";
import createErrorStyle from "./lib/colors/error";
import createInfoStyle from "./lib/colors/info";

import createThinStyle from "./lib/views/thin";
import createLinkStyle from "./lib/views/link";

export default ({ borderRadius, spacing, colors, palette, transitions, }) => {
  const getHeight = (compact = false) => {
    const multiplier = compact ? 3 : 4;
    return `${spacing.unit * multiplier}px`;
  };

  return {
    root: {
      display: "inline-flex",
      flexWrap: "nowrap",
      alignItems: "center",
      justifyContent: "center",
      verticalAlign: "middle",

      maxWidth: "100%",
      height: getHeight(),
      padding: [0, spacing.unit,],
      margin: 0,

      borderRadius,
      border: "none",
      outline: "none !important",

      fontSize: "inherit",
      lineHeight: getHeight(),
      textDecoration: "none",
      whiteSpace: "nowrap",
      color: colors.n400,

      transition: transitions.create(["all",], "shortest"),

      "&:hover": {
        cursor: "pointer",
      },

      "&:active": {
        boxShadow: "none",
        textDecoration: "none",
      },

      "&:disabled": {
        pointerEvents: "none",
        opacity: 0.4,
      },
    },

    compact: {
      height: getHeight(true),
      lineHeight: getHeight(true),
    },

    inline: {
      lineHeight: "inherit",
      padding: 0,
      height: "auto",

      "& $content": {
        margin: 0,
      },
    },

    block: {
      width: "100%",
    },

    selected: {
      transition: "none",
    },

    content: {
      margin: [0, spacing.unit / 2,],
    },

    ...createDefaultStyle(palette, colors),
    ...createPrimaryStyle(palette, colors),
    ...createWarningStyle(palette, colors),
    ...createSuccessStyle(palette, colors),
    ...createErrorStyle(palette, colors),
    ...createInfoStyle(palette, colors),

    ...createThinStyle(palette, colors),
    ...createLinkStyle(palette, colors),
  };
};
