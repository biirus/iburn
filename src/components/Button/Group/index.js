import React, { Children, cloneElement } from "react";
import cn from "classnames";
import PropTypes from "prop-types";

// enhancers
import withStyles from "react-jss";

// styles
import styles from "./styles";

class ButtonGroup extends React.Component {
  static defaultProps = {
    selectable: false,
    justified: false,
  };

  static propTypes = {
    /**
     * Class name string to be merged to the root node
     */
    className: PropTypes.string,
    /**
     * [JSS](http://cssinjs.org/react-jss/?v=v8.5.1) classes object notation
     */
    classes: PropTypes.object,
    /**
     * If it is set to `true`, you will be able to select one of the button.
     * If selected button is changed then `onChange` callback will be called.
     * To define preselected button, you can use the `value` component prop.
     */
    selectable: PropTypes.bool,
    /**
     * It stretches the component to the full width, and all children will be a same size.
     */
    justified: PropTypes.bool,
    /**
     * The color of child `Button` element. It supports the theme meaningfull values
     */
    color: PropTypes.oneOf([
      "default",
      "primary",
      "warning",
      "error",
      "success",
      "info",
    ]),
    /**
     * Array of `<Button />` elements
     */
    children: PropTypes.arrayOf(PropTypes.element).isRequired,
    /**
     * The callback which will be called when `selectable` prop is passed.
     */
    onChange: PropTypes.func,
  };

  handleClick = value => e => {
    const { value: currentValue, onChange } = this.props;

    if (value !== currentValue) {
      this.setState({ value });

      e.persist();

      e.target = this.root;
      e.target.name = this.props.name;
      e.target.value = value;

      onChange && onChange(e);
    }
  };

  getItemProps = child => {
    const { value: currentValue, color, classes, selectable } = this.props;
    const { value } = child.props;

    let props = {
      className: classes.item,
      onClick: this.handleClick(child.props.value),
    };

    if (color) {
      props.color = color;
    }

    if (selectable && currentValue !== undefined) {
      props.selected = value === currentValue;
    }

    return props;
  };

  render () {
    const {
      className,
      justified,
      selectable,
      selected,
      classes,
      children,
      theme,
      ...rest
    } = this.props;

    const classNames = cn(classes.root, className, {
      [classes.justified]: justified,
    });

    return (
      <span {...rest} className={classNames} ref={el => (this.root = el)}>
        {Children.map(children, (child, index) => {
          return (
            <span key={index} className={classes.itemWrapper}>
              {cloneElement(child, this.getItemProps(child))}
            </span>
          );
        })}
      </span>
    );
  }
}

const enhance = withStyles(styles);
export default enhance(ButtonGroup);
