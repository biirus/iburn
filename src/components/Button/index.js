import React from "react";
import cn from "classnames";
import PropTypes from "prop-types";

import getComponent from "./lib/component";

// enhancers
import withStyles from "react-jss";

// styles
import styles from "./styles";

const enhance = withStyles(styles);

/**
 * Buttons allow users to take actions, and make choices, with a single tap.
 */
class Button extends React.Component {
  static defaultProps = {
    selected: false,
    color: "default",
  };

  static propTypes = {
    /**
     * Class name string to be merged to the root node
     */
    className: PropTypes.string,
    /**
     * [JSS](http://cssinjs.org/react-jss/?v=v8.5.1) classes object notation
     */
    classes: PropTypes.object,
    /**
     * Link `href` attribute. If is set `Button` will be redered as `a` element.
     */
    href: PropTypes.string,
    /**
     * If `true` button will have a `selected` styles.
     * This prop is helpfull to deal with Dropdowns or ButtonGroups
     */
    selected: PropTypes.bool,
    /**
     * The padding of a button.
     * It can be a full width of it's parent, inlined with text and have a dense paddings.
     */
    padding: PropTypes.oneOf(["block", "inline", "compact"]),
    /**
     * The color of component. It supports the theme meaningfull values
     */
    color: PropTypes.oneOf([
      "default",
      "primary",
      "warning",
      "error",
      "success",
      "info",
    ]),
    /**
     * The button appearance variant
     */
    variant: PropTypes.oneOf(["link", "thin"]),
    /**
     * The custom component to render the root node
     */
    component: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.func,
      PropTypes.object,
    ]),
    /**
     * Icon that is placed before label
     */
    icon: PropTypes.node,
    /**
     * Icon that is placed after label
     */
    iconAfter: PropTypes.node,
  };

  render () {
    const {
      className,
      classes,
      selected,
      padding,
      variant,
      color,
      theme,

      children,
      component,
      icon,
      iconAfter,
      ...rest
    } = this.props;

    const classNames = cn(classes.root, className, classes[variant], {
      [classes.selected]: selected,
      [classes.compact]: padding === "compact",
      [classes.inline]: padding === "inline",
      [classes.block]: padding === "block",
      [classes[color]]: color && !["link", "thin"].includes(variant),
    });

    const Component = component || getComponent(this.props);

    return (
      <Component {...rest} className={classNames}>
        {icon && <span className={classes.icon}>{icon}</span>}
        {children && <span className={classes.content}>{children}</span>}
        {iconAfter && <span className={classes.iconAfter}>{iconAfter}</span>}
      </Component>
    );
  }
}

export default enhance(Button);
