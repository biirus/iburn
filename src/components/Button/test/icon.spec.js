import React from "react";

import Icon from "../../Icon";
import Button from "../index";
import styles from "../styles";

const classes = getClasses(styles);
const Component = Button.InnerComponent;

describe("<Button />", () => {
  it("should render the <Icon /> before label", () => {
    const wrapper = shallow(
      <Component classes={classes} icon={<Icon label="lock" />}>
        Button content
      </Component>
    );

    expect(wrapper.childAt(0).name()).toEqual("Icon");
    expect(wrapper.childAt(1).type()).toEqual("span");
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the <Icon /> after label", () => {
    const wrapper = shallow(
      <Component classes={classes} iconAfter={<Icon label="lock" />}>
        Button content
      </Component>
    );

    expect(wrapper.childAt(0).type()).toEqual("span");
    expect(wrapper.childAt(1).name()).toEqual("Icon");
    expect(wrapper).toMatchSnapshot();
  });

  it("should render only <Icon />", () => {
    const wrapper = shallow(
      <Component classes={classes} icon={<Icon label="lock" />} />
    );

    expect(wrapper.childAt(0).name()).toEqual("Icon");
    expect(wrapper.prop("children").filter(i => !!i).length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
});
