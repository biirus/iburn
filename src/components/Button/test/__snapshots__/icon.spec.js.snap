// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render only <Icon /> 1`] = `
<StyledButton
  className="root default"
>
  <Icon
    label="lock"
  />
</StyledButton>
`;

exports[`<Button /> should render the <Icon /> after label 1`] = `
<StyledButton
  className="root default"
>
  <span
    className="content"
  >
    Button content
  </span>
  <Icon
    label="lock"
  />
</StyledButton>
`;

exports[`<Button /> should render the <Icon /> before label 1`] = `
<StyledButton
  className="root default"
>
  <Icon
    label="lock"
  />
  <span
    className="content"
  >
    Button content
  </span>
</StyledButton>
`;
